src/shared/unit-file.c

```diff
@@ -21,3 +22,60 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_TIMER,
                       UNIT_PATH);
 }
+
+int unit_validate_alias_symlink_and_warn(const char *filename, const char *target) {
+        const char *src, *dst;
+        UnitType src_type, dst_type;
+        int r, r2;
+
+        /* Check if the *alias* symlink is valid. This applies to symlinks like
+         * /etc/systemd/system/dbus.service → dbus-broker.service, but not to .wants or .requires symlinks
+         * and such. Neither does this apply to symlinks which *link* units, i.e. symlinks to outside of the
+         * unit lookup path.
+         */
+
+        src = basename(filename);
+        dst = basename(target);
+
+        /* Reject cases where the alias or the target unit have instance names. */
+        r = unit_name_to_instance(src, NULL);
```

Comment:
use unit_name_classify()?

Reply:
Right.

Reply:
maybe call this return value src_name_type rather than r? after all you keep it for longer